### PR TITLE
Updated errorreporter to correctly specify application name

### LINKED ISSUE #25262: Error Reporter has application name hard coded to MantidPlot
When reporting an error the `ErrorReporter` current sends `mantidplot` as the application regardless of whether the report came from MantidPlot or MantidWorkbench.

Also, when the report dialog from an unhandled exception mentions MantidPlot twice despite being from the MantidWorkbench
Screenshot from 2019-03-19 21-06-44

### PR DESCRIPTION
**Pull request by Matthew-Andrew**
**Description of work.**

This updates the error-reporter so that it sends the correct aplication name i.e. workbench or mantidplot

This PR also updates the dialog to not refer specifically to mantidplot

**Report to:** @martyngigg  

**To test:**

Unfortunately I think this needs to be tested in packaged versions of linux, OSX and windows builds as the launch scripts for all of them have been changed. For each the following should be checked.

* For workbench and mantidplot check that the errorreport window appears when you run the segfault algorithm. (To run this in workbench run Segfault(False) in the script interpreter).
* Check that the resulting error message contains the correct application name. This can be done either by leaving mantid as it is and checking the errorreport server for the report or changeing the errorreport url in the properties file to something like https://ptsv2.com/s/howitworks.html and checking the body of the message contains the correct application name.

* For workbench and mantidplot check the same things whilst provoking an uncaught exception. Check that the dialog is now non specific
* For mantidplot running `ws = CreateWorkspace(DataX=[0, 1], DataY=[float('inf'), float('inf')])` and trying to plot the resulting workspace should do this.
* For workbench add `raise` before line 88 in workspacewidget.py and then try plotting anything

Fixes #25262  

*This does not require release notes* because it is not a user facing issue

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Comment by martyngigg** [comment]
I'll test it on Linux.

**Comment by TTitcombe** [comment]
I'll test on windows

**Comment by gvardany** [comment]
I can do on :macOS:

**Comment by gvardany** [comment]
Is running Segfault not enough to provoke this?

**Comment by martyngigg** [comment]
@gvardany There are 2 parts. `Segfault` should be enough to test the reporter in the case of a segfault but there is also the case of an unhandled exception that can also raise it. Are you not seeing it all?

File: buildconfig/CMake/Packaging/launch_mantidplot.sh.in [hunk 1]
```diff
@@ -26,4 +26,4 @@ LD_PRELOAD=${LOCAL_PRELOAD} TCMALLOC_RELEASE_RATE=${TCM_RELEASE} \
     TCMALLOC_LARGE_ALLOC_REPORT_THRESHOLD=${TCM_REPORT} \
     PYTHONPATH=${LOCAL_PYTHONPATH} \
     @WRAPPER_PREFIX@$VGLRUN $GDB $INSTALLDIR/bin/@MANTIDPLOT_EXEC@ "$@"@WRAPPER_POSTFIX@ \
-    || PYTHONPATH=${LOCAL_PYTHONPATH} @PYTHON_EXECUTABLE@ @SCRIPTSDIR@/@ERROR_CMD@
+    || PYTHONPATH=${LOCAL_PYTHONPATH} @PYTHON_EXECUTABLE@ @SCRIPTSDIR@/@ERROR_CMD@ --application=mantidplot
```
Comment (DTasev) [inline buildconfig/CMake/Packaging/launch_mantidplot.sh.in]: This and `launch_mantidworkbench.sh` seems to be missing the `--exitcode` param, is this intended?

Comment (Mantid-Matthew) [reply buildconfig/CMake/Packaging/launch_mantidplot.sh.in]: I think that should be set from here https://github.com/mantidproject/mantid/blob/master/buildconfig/CMake/LinuxPackageScripts.cmake#L208 .

Comment (DTasev) [reply buildconfig/CMake/Packaging/launch_mantidplot.sh.in]: 👍 	Didn't see it on the diff. Sorry about that

**Comment by martyngigg** [comment]
This looks good on Linux. The reports for both error cases are sent with the correct application name. :+1:

**Comment by TTitcombe** [comment]
Error reported doesn't show on MantidPlot when running Segfault. The reporter appear for Workbench, but @gemmaguest has checked the server for me and the messages aren't appearing

**Comment by martyngigg** [comment]
@TTitcombe Is it from an installed package? In the first-time setup dialog is the box ticked for reporting usage?

**Comment by TTitcombe** [comment]
It's the installed package and reporting usage is checked. I think @DTasev is looking into it as its similar to an issue he worked on recently

**Comment by gvardany** [comment]
How can we check the error report server?

**Comment by TTitcombe** [comment]
Works on Windows :+1:

**Comment by TTitcombe** [comment]
@gvardany Not sure who has access but I assume @martyngigg and @Mantid-Matthew. If you provide a name for the error reporter which includes mantidplot or workbench so they know what the error message should say, they can check it

File: buildconfig/CMake/Packaging/osx/MantidWorkbench_osx_launcher [hunk 2]
```diff
@@ -3,4 +3,4 @@
 INSTALLDIR=$(cd "$(dirname "$0")"; pwd)
 PLUGIN_DIR=$INSTALLDIR/../PlugIns
 cd $INSTALLDIR
-QT_PLUGIN_PATH=$PLUGIN_DIR ./workbench-script $* || /usr/bin/python ../../scripts/ErrorReporter/error_dialog_app.py --exitcode=$? --directory=$INSTALLDIR --qtdir=$PLUGIN_DIR
+QT_PLUGIN_PATH=$PLUGIN_DIR ./workbench-script $* || /usr/bin/python ../../scripts/ErrorReporter/error_dialog_app.py --exitcode=$? --directory=$INSTALLDIR --qtdir=$PLUGIN_DIR --application==workbench
```
Comment (martyngigg) [inline buildconfig/CMake/Packaging/osx/MantidWorkbench_osx_launcher]: Extra `=` here makes the application `=workbench`. Really minor but worth fixing I think.

**Review by martyngigg — CHANGES_REQUESTED** [review]

**Comment by gvardany** [comment]
Errors got reported as expected on :OS X: except the extra `=`.

**Review by martyngigg — APPROVED** [review]

**Pull request merged**